python/lsst/obs/subaru/strayLight/yStrayLight.py

```diff
@@ -150,8 +155,16 @@ class SubaruStrayLightData(StrayLightData):
         Full path to a FITS files containing the stray-light model.
     """
 
-    def __init__(self, filename):
-        self._filename = filename
+    @classmethod
+    def readFits(cls, filename, **kwargs):
+        calib = cls()
+
+        with fits.open(filename) as hdulist:
+            calib.ampData = [hdulist[extension].data for extension in range(4)]
```

Comment:
Looks like previously the code use `len(hdulist)` so maybe that should be done here as well?
```python
calib.ampData = [hdu.data for hdu in hdulist]
```
? (I think `hdulist` can be an iterator directly).